src/source-af-packet.c

```diff
@@ -853,6 +856,23 @@ static inline int AFPReadFromRingWaitForPacket(AFPThreadVars *ptv)
     return AFP_READ_OK;
 }
 
+/**
+ * \brief AF packet frame ignore logic
+ *
+ * Given a sockaddr_ll of a frame, use the pkttype_filter to decide if the
+ * frame should be ignored. Protect from undefined behavior if there's ever
+ * a sll_pkttype that would shift by too much. At this point, only outgoing
+ * packets (4) are ignored. The highest value in if_linux.h is PACKET_KERNEL (7),
+ * this extra check is being overly cautious.
+ *
+ * \retval true if the frame should be ignored
+ */
+static inline bool AFPShouldIgnoreFrame(AFPThreadVars *ptv, const struct sockaddr_ll *sll)
+{
+    uint32_t pkttype_bit = sll->sll_pkttype > 31 ? 0UL : BIT_U32(sll->sll_pkttype);
```

Comment:
After looking again, this might bit more obvious as, but let me know how you think about the overall idea @victorjulien .
```
if (unlikely(sll->sll_pkttype > 31))
    return false;

return (ptv->pkttype_filter_mask & pkttype_bit) != 0;
```

Reply:
yeah I like this suggestion